cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -770,32 +770,11 @@ class RawIndirectInstruction extends Node, TRawIndirectInstruction {
   }
 }
 
-private predicate isFullyConvertedArgument(Expr e) {
-  exists(Call call |
-    e = call.getAnArgument().getFullyConverted()
-    or
-    e = call.getQualifier().getFullyConverted()
-  )
-}
-
-private predicate isFullyConvertedCall(Expr e) { e = any(Call call).getFullyConverted() }
-
-/** Holds if `Node::asExpr` should map an some operand node to `e`. */
-private predicate convertedExprMustBeOperand(Expr e) {
-  isFullyConvertedArgument(e)
-  or
-  isFullyConvertedCall(e)
-}
-
 /** Holds if `node` is an `OperandNode` that should map `node.asExpr()` to `e`. */
 predicate exprNodeShouldBeOperand(OperandNode node, Expr e) {
-  exists(Operand operand |
-    node.getOperand() = operand and
-    e = operand.getDef().getConvertedResultExpression()
-  |
-    convertedExprMustBeOperand(e)
-    or
-    node.(IndirectOperand).isIRRepresentationOf(_, _)
+  exists(Instruction def |
+    unique( | | getAUse(def)) = node.getOperand() and
```

Comment:
What's the reason for the `unique`?

Reply:
We don't want the `Expr -> Node` direction to be a one-to-many relation. If there happens to be more than one use of the operand, then we'll fall back to picking the `InstructionNode` with the defining `Instruction` as the node for the expression.

Reply:
I guess this was implicitly achieved before by the much more restrictive definition of this predicate?

Reply:
Yeah, exactly. Previously we'd only map an expression to an operand when it was an argument of (or a result of) a function call. And in both cases (at least the argument case) there should always be a unique use.

Reply:
Thanks.

Reply:
One further question about this: so the changes in the test results from, e.g., `call to getenv` to `Call` are due to the result of the call having multiple uses?

Reply:
Yeah, I think so. When there's a unique use this predicate kicks in and ensures that that the `Call` expression is mapped to the dataflow node (and the `Call` expression has a pretty `toString` like `call to getenv`). And when there isn't a unique use we get the `Instruction`'s `Opcode`'s `toString` [which is a lot less user-friendly](https://github.com/github/codeql/blob/5157d4df7b66ddd259d2afe96420eea262af635d/cpp/ql/lib/semmle/code/cpp/ir/implementation/Opcode.qll#L922).

Reply:
Yeah, that was on my list to fix anyway, so no need to solve it here.

Reply:
^ I think this is one of the issues we should tackle as part of the "prettify dataflow paths" issue 😄.

Reply:
Yup, the prettifying issue that was what I was referring to 😄